packages/core-data/src/reducer.js

```diff
@@ -238,6 +238,28 @@ export function userPermissions( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer returning autosaves keyed by their parent's post id.
+ *
+ * @param  {Object} state  Current state.
+ * @param  {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function autosaves( state = {}, action ) {
+	switch ( action.type ) {
+		case 'RECEIVE_AUTOSAVE':
+			const { postId, autosave } = action;
+
+			return {
+				[ postId ]: autosave,
```

Comment:
The REST API schema implies that a post may have multiple autosaves, but in this implementation (and with `getAutosave`) we seem to assume one autosave per post. Who's right?

Reply:
Looks like it's one per author:
https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php#L332

I did some testing and the endpoint can return multiple autosave for a post when edited by multiple users. I'll also have to key by user id, and add that as a parameter to the selector. 

I'm a bit concerned about lack of usability of the selector within the Gutenberg codebase. 

I wonder if it's worth considering keeping the deprecated selectors/actions in the editor module as proxies?

Reply:
Struggling to see how I can determine the user id within Gutenberg ...

Reply:
> Struggling to see how I can determine the current user's id within Gutenberg ...
> 
> I think I'll need that when using the selector from the editor package.

There is a `/wp/v2/users/me` endpoint which could be made to be part of a selector / resolver pair (or built-in entity).

That seems the purest approach, though arguably a bit wasteful when considering the user ID is more readily available via something like the suffix of the `WP_DATA_USER_` localStorage persistence key (managed by data module), or `user_ID` hidden input whenn meta boxes are present.

Reply:
Thanks for the details. It looks like `users/me` might be the best option. On testing it looks like there can be multiple `WP_DATA_USER_` plugins if two users use the same browser.